plugin/completion/base_complete.py

```diff
@@ -120,16 +120,20 @@ def init(self, view, settings):
 
         """
         if not view:
+            log.critical(" no view to initialize completer!")
             return
         current_dir = path.dirname(view.file_name())
         search_scope = SearchScope(
             from_folder=current_dir,
             to_folder=settings.project_base_folder)
         self.flags_manager = FlagsManager(
-            use_cmake=settings.generate_flags_with_cmake,
-            flags_update_strategy=settings.cmake_flags_priority,
-            cmake_prefix_paths=settings.cmake_prefix_paths,
+            view=view,
+            settings=settings,
+            compiler_variant=self.compiler_variant,
```

Comment:
It's quite a serious problem IMO as that means that the views will get flags from most-recently created FlagsManager rather than the one that was created for given view.

The problem existed before, but it wasn't so serious as it only affected .clang-complete file. Now it affects all flags generated for the view. So for example switching between c++ and c files is probably broken right now as those will get the same flags.